core/src/main/java/hudson/PluginManager.java

```diff
@@ -1341,41 +1343,55 @@ public static boolean isNonMetaLabel(String label) {
     
     @Restricted(NoExternalUse.class)
     public HttpResponse doPluginsSearch(@QueryParameter String query, @QueryParameter Integer limit) {
+        final String _query = query.toLowerCase(Locale.ROOT);
+        final boolean isWarningsQuery = _query.contains("warning:");
         List<JSONObject> plugins = new ArrayList<>();
         for (UpdateSite site : Jenkins.get().getUpdateCenter().getSiteList()) {
             List<JSONObject> sitePlugins = site.getAvailables().stream()
                 .filter(plugin -> {
-                    if (StringUtils.isBlank(query)) {
+                    // preserve an existing feature for the "warning:" query
+                    if (isWarningsQuery && plugin.hasWarnings()) {
```

Comment:
💡 That should have been documented in the page directly the first time, I suspect not a lot of people are looking into the code to understand how a search is working :(

Reply:
why? the search should be self explanatory, this was an experts feature put in mostly for @daniel-beck to maintain a feature he already used to list all plugins available that have security warnings